src/core/include/openvino/core/model.hpp

```diff
@@ -153,7 +153,7 @@ class OPENVINO_API Model : public std::enable_shared_from_this<Model> {
     const ov::element::Type& get_output_element_type(size_t i) const;
 
     /// Return the shape of element i
-    const Shape& get_output_shape(size_t i) const;
+    Shape get_output_shape(size_t i) const;
```

Comment:
We should not change return value of these methods. Just deprecate them.
Please add deprecated annotation to all these methods like here:
https://github.com/openvinotoolkit/openvino/blob/2c9953320a575f24856cb5273d6d23c7600c880b/src/core/include/openvino/op/concat.hpp#L40
Release to remove, I suppose, is 2026.0
+
Add OPENVINO_SUPPRESS_DEPRECATED_START/OPENVINO_SUPPRESS_DEPRECATED_END in every place, where these methods are used.